tests/bsim/bluetooth/mesh/src/friendship_common.c

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "friendship_common.h"
+#include "mesh_test.h"
+
+#define LOG_MODULE_NAME friendship_common
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+
+uint16_t bt_mesh_test_friend_lpn_addr;
+ATOMIC_DEFINE(bt_mesh_test_state, BT_MESH_TEST_FRIENDSHIP_FLAGS);
+struct k_sem bt_mesh_test_friendship_events[BT_MESH_TEST_FRIENDSHIP_FLAGS];
+
+void evt_signal(enum bt_mesh_test_friendship_evt_flags evt)
+{
+	atomic_set_bit(bt_mesh_test_state, evt);
+	k_sem_give(&bt_mesh_test_friendship_events[evt]);
+}
+
+int bt_mesh_test_friendship_evt_wait(enum bt_mesh_test_friendship_evt_flags evt,
+				     k_timeout_t timeout)
+{
+	return k_sem_take(&bt_mesh_test_friendship_events[evt], timeout);
```

Comment:
If a variable is global and available everywhere there is no reason to wrap up the single semaphore call into a separate function. It is multiplying logical entities without functionality.

If you need to provide an idea or reason for semaphore usage it is sufficient just add text comments in the code rather than create a separate function.